### PR TITLE
Core/Guild: Implement guild master dethrone feature

### PR DESCRIPTION
**Pull request by roc13x**
**Changes proposed**:

- Implement CMSG_GUILD_REPLACE_GUILD_MASTER
- Allows replacing a guild master inactive for over 90 days, if you have a high enough guild rank.
- Use transaction for GM replacement process

**Issues addressed**: Fixes GM dethrone feature

**Tests performed**: Built and tested ingame

**Known issues and TODO list**: None

### CODE REVIEW

**Comment by funjoker** [comment]
Can you use the newer packet system?

**Comment by roc13x** [comment]
Done ✔️

File: src/server/game/Guilds/Guild.cpp [hunk 1]
```diff
@@ -3330,12 +3344,15 @@ bool Guild::_ModifyBankMoney(CharacterDatabaseTransaction& trans, uint64 amount,
...
 }
 
-void Guild::_SetLeaderGUID(Member* pLeader)
+void Guild::_SetLeader(CharacterDatabaseTransaction& trans, Member* pLeader)
 {
     if (!pLeader)
         return;
 
-    CharacterDatabaseTransaction trans = CharacterDatabase.BeginTransaction();
+    bool isInTransaction = bool(trans);
```
Comment (Ovahlord) [inline src/server/game/Guilds/Guild.cpp:3352]: I'd rather go with  trans != nullptr than casting like this.

Comment (TranceAccount) [reply src/server/game/Guilds/Guild.cpp:3352]: Doesn't `bool(trans)` convert to false if it is a nullptr by definition? 
https://stackoverflow.com/questions/17772103/can-i-use-if-pointer-instead-of-if-pointer-null/17772117#17772117

**Pull request merged**